physical/azure/azure_test.go

```diff
@@ -54,64 +56,52 @@ func TestAzureBackend(t *testing.T) {
 		"accountName":  accountName,
 		"accountKey":   accountKey,
 		"environment":  environmentName,
-		"arm_endpoint": environmentUrl,
+		"arm_endpoint": environmentURL,
 	}, logger)
 
-	defer func() {
-		blobService := cleanupClient.GetBlobService()
-		container := blobService.GetContainerReference(name)
-		container.DeleteIfExists(nil)
-	}()
-
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
 
-	physical.ExerciseBackend(t, backend)
-	physical.ExerciseBackend_ListPrefix(t, backend)
+	return containerURL, backend
 }
 
-func TestAzureBackend_ListPaging(t *testing.T) {
+func TestAzureBackend(t *testing.T) {
 	if os.Getenv("AZURE_ACCOUNT_NAME") == "" ||
 		os.Getenv("AZURE_ACCOUNT_KEY") == "" {
 		t.SkipNow()
 	}
 
-	accountName := os.Getenv("AZURE_ACCOUNT_NAME")
-	accountKey := os.Getenv("AZURE_ACCOUNT_KEY")
-	environmentName := os.Getenv("AZURE_ENVIRONMENT")
-	environmentUrl := os.Getenv("AZURE_ARM_ENDPOINT")
+	containerURL, backend := testFixture(t)
 
-	ts := time.Now().UnixNano()
-	name := fmt.Sprintf("vault-test-%d", ts)
+	defer func() {
+		ctx := context.Background()
+		blobService, _ := containerURL.GetProperties(ctx, azblob.LeaseAccessConditions{})
+		if blobService.StatusCode() == 200 {
+			_, _ = containerURL.Delete(ctx, azblob.ContainerAccessConditions{})
+		}
+	}()
 
-	cleanupEnvironment, err := environmentForCleanupClient(environmentName, environmentUrl)
-	if err != nil {
-		t.Fatalf("err: %s", err)
+	physical.ExerciseBackend(t, backend)
```

Comment:
I'm experiencing an error in here claiming that the blob does not exist:
```
    azure_test.go:85: idempotent delete: -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /Users/calvin/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42
        ===== RESPONSE ERROR (ServiceCode=BlobNotFound) =====
        Description=The specified blob does not exist.
        RequestId:e723de6b-a01e-0074-8090-93ab1d000000
```

Reply:
See comment on `Delete`